jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java

```diff
@@ -1661,24 +1661,13 @@ else if (handler instanceof ServletHandler)
             setServletHandler((ServletHandler)handler);
         else
         {
-            Singleton tail = handler.getTail();
+            // We cannot call super.insertHandler here, because it uses this.setHandler
+            // which gives a warning.  This is the same code, but uses super.setHandler
```

Comment:
Why does `ServletContextHandler.setHandler()` gives a warning?
Seems natural to me that it would set the "child" Handler, which for this class is after the `ServletHandler`, which is always present.

Reply:
Because an ee9 ServletContextHandler is a nested ContextHandler (old API) that contains a core ContextHandler(new API) and sets up a little private chain:  coreContextHandler->core2nestedHandler->ServletContextHandler.
If we allow a setHandler(coreHandler) on core ContextHandler, then it breaks this chain.  We can insert core Handlers between coreContextHandler and core2nestedHandler, but we cannot replace the core2nestHandler without breaking the class.
We can however do a setHandler(nestedHandler) on the ServletContextHandler.

Reply:
Trying new semantic as suggested by @sbordet that setHandler does a set on the ServletHandler